In [165]:
# Dependencies
import pandas as pd

# Files to Load (with path)
school_data_to_load = "Resources/schools_complete.csv"
student_data_to_load = "Resources/students_complete.csv"

# Read School and Student Data File and store into Pandas Data Frames
school_data = pd.read_csv(school_data_to_load)
student_data = pd.read_csv(student_data_to_load)

# Combine the data into a single dataset
school_data_complete = pd.merge(student_data, school_data, how="left", on=["school_name", "school_name"])

# display merged dataset
school_data_complete.head()

Student ID       student_name gender grade        school_name  \
0           0       Paul Bradley      M   9th  Huang High School   
1           1       Victor Smith      M  12th  Huang High School   
2           2    Kevin Rodriguez      M  12th  Huang High School   
3           3  Dr. Richard Scott      M  12th  Huang High School   
4           4         Bonnie Ray      F   9th  Huang High School   

   reading_score  math_score  School ID      type  size   budget  
0             66          79          0  District  2917  1910635  
1             94          61          0  District  2917  1910635  
2             90          60          0  District  2917  1910635  
3             67          58          0  District  2917  1910635  
4             97          84          0  District  2917  1910635

## District Summary

* Calculate the total number of schools

* Calculate the total number of students

* Calculate the total budget

* Calculate the average math score 

* Calculate the average reading score

* Calculate the overall passing rate (overall average score), i.e. (avg. math score + avg. reading score)/2

* Calculate the percentage of students with a passing math score (70 or greater)

* Calculate the percentage of students with a passing reading score (70 or greater)

* Create a dataframe to hold the above results

* Optional: give the displayed data cleaner formatting

In [166]:
# total schools 
tot_sch =len(school_data_complete['school_name'].unique())

# total students 
tot_stu =school_data_complete['student_name'].count()

# total budget 
tot_bud =school_data_complete.groupby(['school_name'])['budget'].mean().sum()

# average math score 
ave_math =school_data_complete['math_score'].mean()

# average read score 
ave_read =school_data_complete['reading_score'].mean()

# average overall (math & read) score  
ave_overall =(ave_math + ave_read)/2

# % of students passing math 
stu_passmath =len(school_data_complete.loc[school_data_complete['math_score'] >= 70.0])
pcnt_passmath =stu_passmath/tot_stu

# % of students passing reading 
stu_passread =len(school_data_complete.loc[school_data_complete['reading_score'] >= 70.0])
pcnt_passread =stu_passread/tot_stu

In [167]:
# Create a dataframe to hold the above results
district_summary = pd.DataFrame({'No. Schools': tot_sch,
                                'No. Students': tot_stu,
                                'Total Budget': tot_bud,
                                'Avg Math Score':ave_math,
                                'Avg Read Score': ave_read,
                                'Avg Overall Score' : ave_overall,
                                '% Stu Pass Math' : pcnt_passmath*100,
                                '% Stu Pass Read': pcnt_passread*100}, index=[0])

# Give the displayed data cleaner formatting
district_summary['Avg Math Score'] = district_summary['Avg Math Score'].map("{:.2f}".format)
district_summary['Avg Read Score'] = district_summary['Avg Read Score'].map("{:.2f}".format)
district_summary['Avg Overall Score'] = district_summary['Avg Overall Score'].map("{:.2f}".format)
district_summary['% Stu Pass Math'] = district_summary['% Stu Pass Math'].map("{:.2f}%".format)
district_summary['% Stu Pass Read'] = district_summary['% Stu Pass Read'].map("{:.2f}%".format)
district_summary['No. Students'] = district_summary['No. Students'].map("{:20,.0f}".format)  
district_summary['Total Budget'] = district_summary['Total Budget'].map("${:20,.0f}".format)  

#===========================================
# display District Summary table

district_summary

No. Schools          No. Students           Total Budget Avg Math Score  \
0           15                39,170  $          24,649,428          78.99   

  Avg Read Score Avg Overall Score % Stu Pass Math % Stu Pass Read  
0          81.88             80.43          74.98%          85.81%

## School Summary

* Create an overview table that summarizes key metrics about each school, including:
  * School Name
  * School Type
  * Total Students
  * Total School Budget
  * Per Student Budget
  * Average Math Score
  * Average Reading Score
  * % Passing Math
  * % Passing Reading
  * Overall Passing Rate (Average of the above two)
  
* Create a dataframe to hold the above results

In [168]:
# Using GroupBy to separate the data into fields according to "school name" & then "type" values
#gschool_data = school_data_complete.groupby(['school_name', 'type'])
gschool_data = school_data_complete.groupby(['school_name'])

# Total Students
stu_persch =gschool_data['Student ID'].count()

# Total School Budget
bud_persch =gschool_data['budget'].mean()

# Per Student Budget
bud_perstu =bud_persch/stu_persch

# Average Math Score
math_scoreave =gschool_data['math_score'].mean()

# Average Reading Score
read_scoreave =gschool_data['reading_score'].mean()

# % Passing Math  
tmp = school_data_complete.loc[school_data_complete['math_score'] >= 70.0, :]
gtmp = tmp.groupby(['school_name'])
perc_mathpass = 100* gtmp['Student ID'].count()/gschool_data['Student ID'].count()

# % Passing Reading
tmp = school_data_complete.loc[school_data_complete['reading_score'] >= 70.0, :]
gtmp = tmp.groupby(['school_name'])
perc_readpass =100* gtmp['Student ID'].count()/gschool_data['Student ID'].count()

# Overall Passing Rate (Average of the above two)
over_percpass =(perc_mathpass + perc_readpass)/2


In [169]:
# =======================================  Create a dataframe to hold the above results
summary_table = pd.DataFrame({'Students': stu_persch,
                            'Budget': bud_persch,
                            'Budget p/Stu': bud_perstu,
                             'Ave Math':math_scoreave,
                            'Ave Read': read_scoreave,
                             '% Math Pass' : perc_mathpass,
                             '% Read Pass' :perc_readpass,
                             '% Pass': over_percpass})
summary_table

Students   Budget  Budget p/Stu   Ave Math   Ave Read  \
school_name                                                                    
Bailey High School         4976  3124928         628.0  77.048432  81.033963   
Cabrera High School        1858  1081356         582.0  83.061895  83.975780   
Figueroa High School       2949  1884411         639.0  76.711767  81.158020   
Ford High School           2739  1763916         644.0  77.102592  80.746258   
Griffin High School        1468   917500         625.0  83.351499  83.816757   
Hernandez High School      4635  3022020         652.0  77.289752  80.934412   
Holden High School          427   248087         581.0  83.803279  83.814988   
Huang High School          2917  1910635         655.0  76.629414  81.182722   
Johnson High School        4761  3094650         650.0  77.072464  80.966394   
Pena High School            962   585858         609.0  83.839917  84.044699   
Rodriguez High School      3999  2547363         637.0  76.842711  80.744686   
Shelton High School        1761  1056600         600.0  83.359455  83.725724   
Thomas High School         1635  1043130         638.0  83.418349  83.848930   
Wilson High School         2283  1319574         578.0  83.274201  83.989488   
Wright High School         1800  1049400         583.0  83.682222  83.955000   

                       % Math Pass  % Read Pass     % Pass  
school_name                                                 
Bailey High School       66.680064    81.933280  74.306672  
Cabrera High School      94.133477    97.039828  95.586652  
Figueroa High School     65.988471    80.739234  73.363852  
Ford High School         68.309602    79.299014  73.804308  
Griffin High School      93.392371    97.138965  95.265668  
Hernandez High School    66.752967    80.862999  73.807983  
Holden High School       92.505855    96.252927  94.379391  
Huang High School        65.683922    81.316421  73.500171  
Johnson High School      66.057551    81.222432  73.639992  
Pena High School         94.594595    95.945946  95.270270  
Rodriguez High School    66.366592    80.220055  73.293323  
Shelton High School      93.867121    95.854628  94.860875  
Thomas High School       93.272171    97.308869  95.290520  
Wilson High School       93.867718    96.539641  95.203679  
Wright High School       93.333333    96.611111  94.972222

## Top Performing Schools (By Passing Rate)

* Sort and display the top five schools in overall passing rate

In [170]:
summary_table.sort_values(['% Pass'], axis=0, ascending=False, inplace=True)
summary_table.head(5) 

Students   Budget  Budget p/Stu   Ave Math   Ave Read  \
school_name                                                                  
Cabrera High School      1858  1081356         582.0  83.061895  83.975780   
Thomas High School       1635  1043130         638.0  83.418349  83.848930   
Pena High School          962   585858         609.0  83.839917  84.044699   
Griffin High School      1468   917500         625.0  83.351499  83.816757   
Wilson High School       2283  1319574         578.0  83.274201  83.989488   

                     % Math Pass  % Read Pass     % Pass  
school_name                                               
Cabrera High School    94.133477    97.039828  95.586652  
Thomas High School     93.272171    97.308869  95.290520  
Pena High School       94.594595    95.945946  95.270270  
Griffin High School    93.392371    97.138965  95.265668  
Wilson High School     93.867718    96.539641  95.203679

## Bottom Performing Schools (By Passing Rate)

* Sort and display the five worst-performing schools

In [171]:
summary_table.tail(5) 

Students   Budget  Budget p/Stu   Ave Math   Ave Read  \
school_name                                                                    
Ford High School           2739  1763916         644.0  77.102592  80.746258   
Johnson High School        4761  3094650         650.0  77.072464  80.966394   
Huang High School          2917  1910635         655.0  76.629414  81.182722   
Figueroa High School       2949  1884411         639.0  76.711767  81.158020   
Rodriguez High School      3999  2547363         637.0  76.842711  80.744686   

                       % Math Pass  % Read Pass     % Pass  
school_name                                                 
Ford High School         68.309602    79.299014  73.804308  
Johnson High School      66.057551    81.222432  73.639992  
Huang High School        65.683922    81.316421  73.500171  
Figueroa High School     65.988471    80.739234  73.363852  
Rodriguez High School    66.366592    80.220055  73.293323

## Math Scores by Grade

* Create a table that lists the average Reading Score for students of each grade level (9th, 10th, 11th, 12th) at each school.

* Create a pandas series for each grade. Hint: use a conditional statement.

* Group each series by school

* Combine the series into a dataframe

* Optional: give the displayed data cleaner formatting

In [172]:
tmp = school_data_complete.loc[school_data_complete['grade'] =='9th', :]
gtmp09 = tmp.groupby(['school_name'])['math_score'].mean()
tmp = school_data_complete.loc[school_data_complete['grade'] =='10th', :]
gtmp10 = tmp.groupby(['school_name'])['math_score'].mean()
tmp = school_data_complete.loc[school_data_complete['grade'] =='11th', :]
gtmp11 = tmp.groupby(['school_name'])['math_score'].mean()
tmp = school_data_complete.loc[school_data_complete['grade'] =='12th', :]
gtmp12 = tmp.groupby(['school_name'])['math_score'].mean()

# =======================================  Create a dataframe to hold the above results
math_bygrade = pd.DataFrame({'Gr9th Math Score': gtmp09,
                            'Gr10th Math Score': gtmp10,
                            'Gr11th Math Score': gtmp11,
                            'Gr12th Math Score': gtmp12 })

# give the displayed data cleaner formatting
math_bygrade['Gr9th Math Score'] = math_bygrade['Gr9th Math Score'].map("{:.2f}".format)
math_bygrade['Gr10th Math Score'] = math_bygrade['Gr10th Math Score'].map("{:.2f}".format)
math_bygrade['Gr11th Math Score'] = math_bygrade['Gr11th Math Score'].map("{:.2f}".format)
math_bygrade['Gr12th Math Score'] = math_bygrade['Gr12th Math Score'].map("{:.2f}".format)

math_bygrade

Gr9th Math Score Gr10th Math Score Gr11th Math Score  \
school_name                                                                  
Bailey High School               77.08             77.00             77.52   
Cabrera High School              83.09             83.15             82.77   
Figueroa High School             76.40             76.54             76.88   
Ford High School                 77.36             77.67             76.92   
Griffin High School              82.04             84.23             83.84   
Hernandez High School            77.44             77.34             77.14   
Holden High School               83.79             83.43             85.00   
Huang High School                77.03             75.91             76.45   
Johnson High School              77.19             76.69             77.49   
Pena High School                 83.63             83.37             84.33   
Rodriguez High School            76.86             76.61             76.40   
Shelton High School              83.42             82.92             83.38   
Thomas High School               83.59             83.09             83.50   
Wilson High School               83.09             83.72             83.20   
Wright High School               83.26             84.01             83.84   

                      Gr12th Math Score  
school_name                              
Bailey High School                76.49  
Cabrera High School               83.28  
Figueroa High School              77.15  
Ford High School                  76.18  
Griffin High School               83.36  
Hernandez High School             77.19  
Holden High School                82.86  
Huang High School                 77.23  
Johnson High School               76.86  
Pena High School                  84.12  
Rodriguez High School             77.69  
Shelton High School               83.78  
Thomas High School                83.50  
Wilson High School                83.04  
Wright High School                83.64

## Reading Score by Grade 

* Perform the same operations as above for reading scores

In [173]:
tmp = school_data_complete.loc[school_data_complete['grade'] =='9th', :]
gtmp09 = tmp.groupby(['school_name'])['reading_score'].mean()
tmp = school_data_complete.loc[school_data_complete['grade'] =='10th', :]
gtmp10 = tmp.groupby(['school_name'])['reading_score'].mean()
tmp = school_data_complete.loc[school_data_complete['grade'] =='11th', :]
gtmp11 = tmp.groupby(['school_name'])['reading_score'].mean()
tmp = school_data_complete.loc[school_data_complete['grade'] =='12th', :]
gtmp12 = tmp.groupby(['school_name'])['reading_score'].mean()

# =======================================  Create a dataframe to hold the above results
read_bygrade = pd.DataFrame({'Gr9th Read Score': gtmp09,
                            'Gr10th Read Score': gtmp10,
                            'Gr11th Read Score': gtmp11,
                            'Gr12th Read Score': gtmp12 })

# give the displayed data cleaner formatting
read_bygrade['Gr9th Read Score'] = read_bygrade['Gr9th Read Score'].map("{:.2f}".format)
read_bygrade['Gr10th Read Score'] = read_bygrade['Gr10th Read Score'].map("{:.2f}".format)
read_bygrade['Gr11th Read Score'] = read_bygrade['Gr11th Read Score'].map("{:.2f}".format)
read_bygrade['Gr12th Read Score'] = read_bygrade['Gr12th Read Score'].map("{:.2f}".format)

read_bygrade

Gr9th Read Score Gr10th Read Score Gr11th Read Score  \
school_name                                                                  
Bailey High School               81.30             80.91             80.95   
Cabrera High School              83.68             84.25             83.79   
Figueroa High School             81.20             81.41             80.64   
Ford High School                 80.63             81.26             80.40   
Griffin High School              83.37             83.71             84.29   
Hernandez High School            80.87             80.66             81.40   
Holden High School               83.68             83.32             83.82   
Huang High School                81.29             81.51             81.42   
Johnson High School              81.26             80.77             80.62   
Pena High School                 83.81             83.61             84.34   
Rodriguez High School            80.99             80.63             80.86   
Shelton High School              84.12             83.44             84.37   
Thomas High School               83.73             84.25             83.59   
Wilson High School               83.94             84.02             83.76   
Wright High School               83.83             83.81             84.16   

                      Gr12th Read Score  
school_name                              
Bailey High School                80.91  
Cabrera High School               84.29  
Figueroa High School              81.38  
Ford High School                  80.66  
Griffin High School               84.01  
Hernandez High School             80.86  
Holden High School                84.70  
Huang High School                 80.31  
Johnson High School               81.23  
Pena High School                  84.59  
Rodriguez High School             80.38  
Shelton High School               82.78  
Thomas High School                83.83  
Wilson High School                84.32  
Wright High School                84.07

## Scores by School Spending

* Create a table that breaks down school performances based on average Spending Ranges (Per Student). Use 4 reasonable bins to group school spending. Include in the table each of the following:
  * Average Math Score
  * Average Reading Score
  * % Passing Math
  * % Passing Reading
  * Overall Passing Rate (Average of the above two)

In [174]:
# looking at Budget stats to decide on bins 
summary_table['Budget p/Stu'].describe()

count     15.000000
mean     620.066667
std       28.544368
min      578.000000
25%      591.500000
50%      628.000000
75%      641.500000
max      655.000000
Name: Budget p/Stu, dtype: float64

In [175]:
# bins
spending_bins = [0, 585, 615, 645, 675]
group_names = ["Low:<$585", "Med:$585-615", "Med-High:$615-645", "High:$645-675"]

summary_table['Budget Bin'] = pd.cut(summary_table['Budget p/Stu'], spending_bins, labels=group_names)

byspending_table  =summary_table.drop(['Students','Budget','Budget p/Stu'], axis = 1) 
byspending_table

Ave Math   Ave Read  % Math Pass  % Read Pass  \
school_name                                                             
Cabrera High School    83.061895  83.975780    94.133477    97.039828   
Thomas High School     83.418349  83.848930    93.272171    97.308869   
Pena High School       83.839917  84.044699    94.594595    95.945946   
Griffin High School    83.351499  83.816757    93.392371    97.138965   
Wilson High School     83.274201  83.989488    93.867718    96.539641   
Wright High School     83.682222  83.955000    93.333333    96.611111   
Shelton High School    83.359455  83.725724    93.867121    95.854628   
Holden High School     83.803279  83.814988    92.505855    96.252927   
Bailey High School     77.048432  81.033963    66.680064    81.933280   
Hernandez High School  77.289752  80.934412    66.752967    80.862999   
Ford High School       77.102592  80.746258    68.309602    79.299014   
Johnson High School    77.072464  80.966394    66.057551    81.222432   
Huang High School      76.629414  81.182722    65.683922    81.316421   
Figueroa High School   76.711767  81.158020    65.988471    80.739234   
Rodriguez High School  76.842711  80.744686    66.366592    80.220055   

                          % Pass         Budget Bin  
school_name                                          
Cabrera High School    95.586652          Low:<$585  
Thomas High School     95.290520  Med-High:$615-645  
Pena High School       95.270270       Med:$585-615  
Griffin High School    95.265668  Med-High:$615-645  
Wilson High School     95.203679          Low:<$585  
Wright High School     94.972222          Low:<$585  
Shelton High School    94.860875       Med:$585-615  
Holden High School     94.379391          Low:<$585  
Bailey High School     74.306672  Med-High:$615-645  
Hernandez High School  73.807983      High:$645-675  
Ford High School       73.804308  Med-High:$615-645  
Johnson High School    73.639992      High:$645-675  
Huang High School      73.500171      High:$645-675  
Figueroa High School   73.363852  Med-High:$615-645  
Rodriguez High School  73.293323  Med-High:$615-645

## Scores by School Size

* Perform the same operations as above, based on school size.

In [176]:
# looking at Student stats to decide on bins 
summary_table['Students'].describe()

count      15.000000
mean     2611.333333
std      1420.915282
min       427.000000
25%      1698.000000
50%      2283.000000
75%      3474.000000
max      4976.000000
Name: Students, dtype: float64

In [177]:
# bins
size_bins = [0, 1000, 2000, 5000]
group_names = ["Small:<1k", "Medium:1k-2k", "Large:2k-5k"]

summary_table['Size Bin'] = pd.cut(summary_table['Students'], size_bins, labels=group_names)
summary_table.head(17)

bysize_table  =summary_table.drop(['Students','Budget','Budget p/Stu','Budget Bin'], axis = 1) 
bysize_table.head(17)

Ave Math   Ave Read  % Math Pass  % Read Pass  \
school_name                                                             
Cabrera High School    83.061895  83.975780    94.133477    97.039828   
Thomas High School     83.418349  83.848930    93.272171    97.308869   
Pena High School       83.839917  84.044699    94.594595    95.945946   
Griffin High School    83.351499  83.816757    93.392371    97.138965   
Wilson High School     83.274201  83.989488    93.867718    96.539641   
Wright High School     83.682222  83.955000    93.333333    96.611111   
Shelton High School    83.359455  83.725724    93.867121    95.854628   
Holden High School     83.803279  83.814988    92.505855    96.252927   
Bailey High School     77.048432  81.033963    66.680064    81.933280   
Hernandez High School  77.289752  80.934412    66.752967    80.862999   
Ford High School       77.102592  80.746258    68.309602    79.299014   
Johnson High School    77.072464  80.966394    66.057551    81.222432   
Huang High School      76.629414  81.182722    65.683922    81.316421   
Figueroa High School   76.711767  81.158020    65.988471    80.739234   
Rodriguez High School  76.842711  80.744686    66.366592    80.220055   

                          % Pass      Size Bin  
school_name                                     
Cabrera High School    95.586652  Medium:1k-2k  
Thomas High School     95.290520  Medium:1k-2k  
Pena High School       95.270270     Small:<1k  
Griffin High School    95.265668  Medium:1k-2k  
Wilson High School     95.203679   Large:2k-5k  
Wright High School     94.972222  Medium:1k-2k  
Shelton High School    94.860875  Medium:1k-2k  
Holden High School     94.379391     Small:<1k  
Bailey High School     74.306672   Large:2k-5k  
Hernandez High School  73.807983   Large:2k-5k  
Ford High School       73.804308   Large:2k-5k  
Johnson High School    73.639992   Large:2k-5k  
Huang High School      73.500171   Large:2k-5k  
Figueroa High School   73.363852   Large:2k-5k  
Rodriguez High School  73.293323   Large:2k-5k

## Scores by School Type

* Perform the same operations as above, based on school type.

In [178]:
# repeating some of the same steps as for "summary_table"  but now adding "type" as 2nd group in groupby
xgschool_data = school_data_complete.groupby(['school_name', 'type'])

# Average Math Score
math_scoreave =xgschool_data['math_score'].mean()

# Average Reading Score
read_scoreave =xgschool_data['reading_score'].mean()

# % Passing Math  
tmp = school_data_complete.loc[school_data_complete['math_score'] >= 70.0, :]
gtmp = tmp.groupby(['school_name'])
perc_mathpass = 100* gtmp['Student ID'].count()/xgschool_data['Student ID'].count()

# % Passing Reading
tmp = school_data_complete.loc[school_data_complete['reading_score'] >= 70.0, :]
gtmp = tmp.groupby(['school_name'])
perc_readpass =100* gtmp['Student ID'].count()/xgschool_data['Student ID'].count()

# Overall Passing Rate (Average of the above two)
over_percpass =(perc_mathpass + perc_readpass)/2

# =======================================  Create a dataframe to hold the above results
xsummary_table = pd.DataFrame({'Ave Math':math_scoreave,
                            'Ave Read': read_scoreave,
                             '% Math Pass' : perc_mathpass,
                             '% Read Pass' :perc_readpass,
                             '% Pass': over_percpass})

xsummary_table.sort_values(['% Pass'], axis=0, ascending=False, inplace=True)
xsummary_table 

Ave Math   Ave Read  % Math Pass  \
school_name           type                                          
Cabrera High School   Charter   83.061895  83.975780    94.133477   
Thomas High School    Charter   83.418349  83.848930    93.272171   
Pena High School      Charter   83.839917  84.044699    94.594595   
Griffin High School   Charter   83.351499  83.816757    93.392371   
Wilson High School    Charter   83.274201  83.989488    93.867718   
Wright High School    Charter   83.682222  83.955000    93.333333   
Shelton High School   Charter   83.359455  83.725724    93.867121   
Holden High School    Charter   83.803279  83.814988    92.505855   
Bailey High School    District  77.048432  81.033963    66.680064   
Hernandez High School District  77.289752  80.934412    66.752967   
Ford High School      District  77.102592  80.746258    68.309602   
Johnson High School   District  77.072464  80.966394    66.057551   
Huang High School     District  76.629414  81.182722    65.683922   
Figueroa High School  District  76.711767  81.158020    65.988471   
Rodriguez High School District  76.842711  80.744686    66.366592   

                                % Read Pass     % Pass  
school_name           type                              
Cabrera High School   Charter     97.039828  95.586652  
Thomas High School    Charter     97.308869  95.290520  
Pena High School      Charter     95.945946  95.270270  
Griffin High School   Charter     97.138965  95.265668  
Wilson High School    Charter     96.539641  95.203679  
Wright High School    Charter     96.611111  94.972222  
Shelton High School   Charter     95.854628  94.860875  
Holden High School    Charter     96.252927  94.379391  
Bailey High School    District    81.933280  74.306672  
Hernandez High School District    80.862999  73.807983  
Ford High School      District    79.299014  73.804308  
Johnson High School   District    81.222432  73.639992  
Huang High School     District    81.316421  73.500171  
Figueroa High School  District    80.739234  73.363852  
Rodriguez High School District    80.220055  73.293323

## Observable trends based on the data


Observable trends are as follows:

* Overall scores are mostly determined by school type - any of the 8 Charter schools show higher overall scores than any of the Disctric schools. The gap is large - worst score for Charter schools is at 94.379391, much larger than best score for District at 74.306672

* School size (in total number of students) seem to impact overall scores - 7 of 8 large size schools are at the bottom. However, when looking closer the large size school ranked well (ranked 4th) is of Charter type. Therefore, type seems to be  more significant than size

* Similar comments apply to Budget size (USD spending per student) - does not seem to correlate with overall score ranking. Actually the top ranked school has a low budget, while all schools with high budgets are on the bottom half of the score ranking   
